src/Components/App/App.test.js

```diff
@@ -0,0 +1,186 @@
+    
+    await waitFor(() => expect(screen.getByTestId('posting-view-element')).toBeInTheDocument());
+    await waitFor(() => screen.getByText('driver').click());
+    await waitFor(() => screen.getByText("Sign me up!").click());
+
+    await waitFor(() => expect(screen.getByTestId('upcoming-1-posting-4')).toBeInTheDocument());
+    await waitFor(() => screen.getByAltText('return-home-button').click())
+
+    await waitFor(() => expect(screen.queryByTestId('posting-view-element')).not.toBeInTheDocument());
+    await waitFor(() => expect(history.location.pathname).toBe("/"));
+  })
+
+  it("When enters keyword in search it finds all events with this keyword", async() => {
+    render(
+      <MemoryRouter>
+        <App />
+      </MemoryRouter>
+    )
+    await waitFor(() => expect(apiCalls.getUser).toHaveBeenCalledTimes(1));
+    await waitFor(() => expect(apiCalls.getPostings).toHaveBeenCalledTimes(1));
+    await waitFor(() => expect(screen.getByTestId("navigation-element")).toBeInTheDocument());
+
+    const userInput = screen.getByPlaceholderText("i.e Boulder...");
+    userEvent.type(userInput, 'Food Delivery');
+
+    const searchbutton = screen.getByText("search");
+    fireEvent.click(searchbutton);
+
+    await waitFor(() => expect(screen.getByText("Food Delivery")).toBeInTheDocument())
+    await waitFor(() => expect(screen.queryByText("Something Else, LLC")).not.toBeInTheDocument())
+  })
+
+  it("When user clicks on a sort button it shows events in desc order by date, by clicking again it shows in asce", async() => {
+    render(
+      <MemoryRouter>
+        <App />
+      </MemoryRouter>
+    )
+    await waitFor(() => expect(apiCalls.getUser).toHaveBeenCalledTimes(1));
+    await waitFor(() => expect(apiCalls.getPostings).toHaveBeenCalledTimes(1));
+    await waitFor(() => expect(screen.getByTestId("navigation-element")).toBeInTheDocument());
+
+    const sortButton = screen.getByText("sort");
+    fireEvent.click(sortButton);
+
+    const allEventCards = await waitFor(() => screen.getAllByTestId("posting-card-element"))
+    expect(allEventCards[0].href === 'http://localhost/postings/event-20')
+    expect(allEventCards[1].href === 'http://localhost/postings/event-2')
+
+    fireEvent.click(sortButton);
+    expect(allEventCards[0].href === 'http://localhost/postings/event-2')
+    expect(allEventCards[1].href === 'http://localhost/postings/event-20')
+  })
+
+  it("When user chooses a category the page displays the events of that category only", async() => {
+    render(
+      <MemoryRouter>
+        <App />
+      </MemoryRouter>
+    )
+    await waitFor(() => expect(apiCalls.getUser).toHaveBeenCalledTimes(1));
+    await waitFor(() => expect(apiCalls.getPostings).toHaveBeenCalledTimes(1));
+    await waitFor(() => expect(screen.getByTestId("navigation-element")).toBeInTheDocument());
+
+    const filteredCategory = screen.getByTestId("select-input");
+    fireEvent.change(filteredCategory, {
```

Comment:
It still puzzles me why this one interference with the single page redirect test...